telethon_generator/tl_generator.py

```diff
@@ -687,21 +687,17 @@ def write_request_result_code(builder, tlobject):
                          will be written
         """
         if tlobject.result.startswith('Vector<'):
-            # Vector results are a bit special since they can also be composed
-            # of integer values and such; however, the result of requests is
-            # not parsed as arguments are and it's a bit harder to tell which
-            # is which.
-            if tlobject.result == 'Vector<int>':
+            # Check if Vector contents are boxed or not
+            # We're implying that constructor names always
+            # start with lowecase and types start with uppercase
+            match = re.match(r'Vector<([\w\d]+)>', tlobject.result)
+            if match and match.group(1).lower():
```

Comment:
By the way, this will convert the string to lower case, which will always be truthy if present (and it will always be present if there's a match). You want `.islower()`.

Reply:
@Lonami lol, indeed